### PR TITLE
Tutorial reworked

### PR DESCRIPTION
**Pull request by comradekingu**
At the intersection of not knowing how to do newlines in lisp, and wanting read the tutorial.

### CODE REVIEW

File: source/tutorial.lisp [hunk 1]
```diff
@@ -6,59 +6,59 @@
...
    (:p "Commands are invoked by pressing specific keys or from
 the " (:code "execute-command") " menu (" (:code (binding-keys 'execute-command))
 ").")
-   (:p "Keybindings are represented like this: 'C-x'. In this example, 'C' is a
-shortcut for the modifier 'control', and 'x' represents the character 'x'. To
-input the 'C-x' keybinding you would keep 'control' pressed and then hit 'x'.
-Multiple key presses can be chained: in 'C-x C-s', you would have to press
-'C-x', and then press 'C-s'.")
-   (:p "Modifier keys legend:")
+   (:p "Key-sequences used for key-bindings are represented like this: 'C-x'.~%
```
Comment (jmercouris) [inline source/tutorial.lisp:13]: I see, I assume you are trying to make a new paragraph here?

if so, you would create a new

```
(:p "my new paragraph")
```

If you are just trying to do a line break within a paragraph

```
(:p "hello tomato"
  (:br)
  "a new line in the paragraph about tomato")
```

**Comment by jmercouris** [comment]
Firstly, thank you very much!

I appreciate some of the changes. I've left a comment about how you can support the new lines :-)

If you don't have time to do it, I can try to apply some of these changes!

**Comment by comradekingu** [comment]
@jmercouris Thank you. Removed those for now, might get there in the end.
Is there anything you dislike for now?

File: source/tutorial.lisp [hunk 2]
```diff
@@ -355,34 +341,34 @@ creatively tailor-made by users to automate their own use of Nyxt.")
...
-         ": Open file in Nyxt or externally.  See " (:code "open-file-function") ".")
+         ": Opens a file in Nyxt or via other means.  Read " (:code "open-file-function") ".")
     (:li (command-markup 'edit-with-external-editor)
-         ": Edit selected HTML input tag with an external editor.")
+         ": Edit selected HTML input tag in an external editor.")
     (:li (command-markup 'quit) ": Close all Nyxt windows and quit."))
 
    (:h2 "The Nyxt help system")
-   (:p "Nyxt provides introspective and help capabilities.  All commands,
+   (:p "Nyxt provides introspection, clarity, and help. All commands,
```
Comment (Ambrevar) [inline source/tutorial.lisp:353]: Why "clarity"?

**Review by Ambrevar — COMMENTED** [review]
Looks good to me beside the small "clarity" nit.
@jmercouris Feel free to merge if you are happy with it.

File: source/tutorial.lisp (hunk 2 above)
Comment (comradekingu) [reply source/tutorial.lisp:353]: It is something in between "introspection" and "help"..

Comment (Ambrevar) [reply source/tutorial.lisp:353]: I am not sure what you mean.  Maybe "Nyxt strives to expose clear documentation"?

File: source/tutorial.lisp [hunk 3]
```diff
@@ -6,59 +6,59 @@
 (defun tutorial-content ()
   (spinneret:with-html-string
    (:h2 "Core concepts")
-   (:h3 "Keybindings and commands")
+   (:h3 "Key-bindings and commands")
```
Comment (aartaka) [inline source/tutorial.lisp:9]: I've seen the spaced writing ([Wikipedia does that](https://en.wikipedia.org/wiki/Keyboard_shortcut#Customization)), the no-space writing (the one we use), but I've never seen the hyphenated writing. What's your reference for this?

Emacs- and VI-related resources use the no-space or spaced writing interchangeably, and that's our point of reference.

The heuristic I personally use for compounded words goes like:
- noun -- write it as one word,
- adjective -- use hyphen,
- verb -- use space.

I may be wrong, though.

File: source/tutorial.lisp [hunk 4]
```diff
@@ -6,59 +6,59 @@
...
    (:p "Commands are invoked by pressing specific keys or from
 the " (:code "execute-command") " menu (" (:code (binding-keys 'execute-command))
 ").")
-   (:p "Keybindings are represented like this: 'C-x'. In this example, 'C' is a
-shortcut for the modifier 'control', and 'x' represents the character 'x'. To
-input the 'C-x' keybinding you would keep 'control' pressed and then hit 'x'.
-Multiple key presses can be chained: in 'C-x C-s', you would have to press
-'C-x', and then press 'C-s'.")
-   (:p "Modifier keys legend:")
+   (:p "Key-sequences used for key-bindings are represented like this: 'C-x'.
```
Comment (aartaka) [inline source/tutorial.lisp:13]: Same as with key-bindings, never seen a hyphenated writing. Any reason for exactly this one?

File: source/tutorial.lisp [hunk 5]
```diff
@@ -6,59 +6,59 @@
...
+Here, 'C' is a
+shortcut for the modifier 'control', and 'x' is the character 'x'. To
+input 'C-x', keep 'control' pressed and then hit 'x'.
+Chain distinct key-sequences, you madlad. With 'C-x C-s', 'C-x', is pressed first,
+and then 'C-s'.")
+   (:p "Modifier-keys legend:")
    (:ul
     (:li (:code "control") " (" (:code "C") "): Control key")
-    (:li (:code "super") " (" (:code "S") "): Windows key, Command key")
+    (:li (:code "super") " (" (:code "S") "): Super key, Command key")
```
Comment (aartaka) [inline source/tutorial.lisp:22]: Windows key is not a nice reference, but a useful one. How about "Super (Windows) key"?

Comment (aartaka) [reply source/tutorial.lisp:22]: Or even all the three options alongside to maximize possible associations.

File: source/tutorial.lisp [hunk 6]
```diff
@@ -6,59 +6,59 @@
...
+    (:li (:code (binding-keys 'nyxt/web-mode:follow-hint)) ": Follow a link in the current buffer")
+    (:li (:code (binding-keys 'nyxt/web-mode:follow-hint-new-buffer)) ": Follow a link in a new buffer")
     (:li (:code (binding-keys 'quit)) ": Quit")
     (:li (:code (binding-keys 'execute-command)) ": Run a command by name")
     (:li (:code (binding-keys 'describe-bindings)) ": List all bindings for the current buffer"))
 
    (:h3 "Buffers")
-   (:p "Nyxt uses the concept of buffers instead of tabs. Unlike tabs, buffers
-are fully separated, each buffer having its own behavior and settings.")
+   (:p "Buffers are instead of tabs. Unlike tabs, buffers
```
Comment (aartaka) [inline source/tutorial.lisp:43]: This feels vague. The idea of naming/concept is important to leave here. I like the old version more, but I'm sure you can find even a better phrasing here to shorten it and retain the necessary meaning :)

File: source/tutorial.lisp [hunk 7]
```diff
@@ -6,59 +6,59 @@
...
    (:h3 "Modes")
-   (:p "Each buffer has its own list of modes, ordered by priority.  A mode is a
+   (:p "Each buffer has its own list of modes, ordered by priority. A mode is a
 set of functions, hooks, keybindings and other facilities that may modify the
-behavior of a buffer.  For example, 'blocker-mode' can be used for domain-based
-adblocking while 'no-script-mode' disables JavaScript.")
-   (:p "Each buffer has separate instances of modes, which means that altering
-the settings of a mode in a buffer does not impact other buffers.  Mode specific
+behaviour of a buffer. For example, 'blocker-mode' can be used for domain-based
+ad-blocking while 'noscript-mode' disables JavaScript.")
```
Comment (aartaka) [inline source/tutorial.lisp:49]: `no-script-mode` is actually the name of the mode, so it should stay with the hyphen.

File: source/tutorial.lisp [hunk 8]
```diff
@@ -88,54 +87,52 @@ upon return.  The suggestion under the cursor is not processed if not marked.")
...
     (:li  (command-markup 'nyxt/prompt-buffer-mode:toggle-attributes-display
                           :modes (list (make-instance 'nyxt/prompt-buffer-mode:prompt-buffer-mode)))
-          ": Change which attributes are displayed in the suggestions list."))
+          ": Change which attributes are displayed in the suggestion list."))
 
    (:h3 "Message area")
-   (:p "The message area represents a space (typically at the bottom of a
-window) where Nyxt outputs messages back to you. To view the history of all
-messages, invoke the command " (:code "list-messages") ".")
+   (:p "Messages are displayed in a area (typically at the bottom of a
```
Comment (aartaka) [inline source/tutorial.lisp:93]: "message area" is actually a term (a stricter version is `message-buffer`). The heading says that, but repeating the term in the body of paragraph wouldn't hurt.

File: source/tutorial.lisp [hunk 9]
```diff
@@ -88,54 +87,52 @@ upon return.  The suggestion under the cursor is not processed if not marked.")
...
-   (:p "The status area is where information about the state of that buffer is
-printed (typically at the bottom of a window). By default, this includes the
-active modes, the URL, and the title of the current buffer.")
+   (:p "The status area (typically at the bottom of a window) displays buffer state.
+By default this includes the active modes, URL, and title of the current buffer.")
 
    (:h2 "Basic controls")
    (:h3 "Moving within a buffer")
-   (:p "To move within a buffer, several commands are provided:")
+   (:p "Commands to navigate buffers:")
```
Comment (aartaka) [inline source/tutorial.lisp:103]: That's quite an ambiguous one. "navigate" is:
- Move around internet.
- Look for something around a set of things.
- And only as a last one, move around.

So I'm not sure it's an equivalent rewriting.

File: source/tutorial.lisp [hunk 10]
```diff
@@ -88,54 +87,52 @@ upon return.  The suggestion under the cursor is not processed if not marked.")
...
     (:li (command-markup 'nyxt/web-mode:paste) ": " (command-docstring-first-sentence 'nyxt/web-mode:paste)))
-   (:p "Passing through webpage's data:")
+   (:p "Passing through a webpage's data:")
    (:ul
     (:li (command-markup 'copy-url) ": " (command-docstring-first-sentence 'copy-url))
     (:li (command-markup 'copy-title) ": " (command-docstring-first-sentence 'copy-title))
     (:li (command-markup 'nyxt/web-mode:copy-placeholder) ": " (command-docstring-first-sentence 'nyxt/web-mode:copy-placeholder))
     (:li (command-markup 'nyxt/web-mode:copy-hint-url) ": " (command-docstring-first-sentence 'nyxt/web-mode:copy-hint-url)))
-   (:p "Leveraging password managers: ")
+   (:p "Leveraging password-managers: ")
```
Comment (aartaka) [inline source/tutorial.lisp:135]: Hyphen?

File: source/tutorial.lisp [hunk 11]
```diff
@@ -145,18 +142,18 @@ to quickly find whatever buffer you are looking for.")
     (:li (command-markup 'nyxt/web-mode:paste-from-clipboard-ring) ": " (command-docstring-first-sentence 'nyxt/web-mode:paste-from-clipboard-ring))
     (:li (command-markup 'copy-system-information) ": " (command-docstring-first-sentence 'copy-system-information)))
    (:h3 "Link navigation")
-   (:p "Link-hinting allows you to visit URLs on a page without using the mouse.
-Invoke one of the commands below: several hints will appear on screen and all
-links on the page will be listed in the prompt buffer.  You can select the hints
-by matching against the hint, the URL or the title.")
+   (:p "Link-hinting allows visiting URLs on a page without mousing.
```
Comment (aartaka) [inline source/tutorial.lisp:145]: Is "mousing" so widespread to not be confusing? It made me stop for a second there :(

File: source/tutorial.lisp [hunk 12]
```diff
@@ -179,34 +175,34 @@ place rather than having to jump around in a buffer (or multiple buffers).")
...
-key-binding " (:code "C-i") " executes " (:code "autofill") ". Suppose
-a user is using their email client which also uses " (:code "C-i") " for the italic command. Thus, after
+renderer. When using the default binding of Nyxt (" (:code "web-cua-map") "), the
+key-binding " (:code "C-i") " executes " (:code "autofill") ". If using an e-mail client
+which also uses " (:code "C-i") " for the italic command, after
 executing " (:code "application-mode") " the " (:code "C-i") " binding is associated
-with the webpage's italic command instead of " (:code "autofill") ". Finally, the
-user can return to their configuration just by executing " (:code "application-mode") " again.")
+with the webpage's italic command instead of " (:code "autofill") ". Return to the
+configuration by executing " (:code "application-mode") " again.")
```
Comment (aartaka) [inline source/tutorial.lisp:203]: Which "configuration"? It's a bit unclear. Maybe "usual key configuration" or something along the lines of it?

File: source/tutorial.lisp [hunk 13]
```diff
@@ -223,80 +219,71 @@ buffer.")
...
          (command-docstring-first-sentence 'nyxt/no-webgl-mode:no-webgl-mode)))
    (:h3 "Visual mode")
-   (:p "Select text without a mouse. Nyxt's "
-       (:code "visual-mode") " imitates Vim's visual mode (and comes with the
-CUA and Emacs-like keybindings out of the box, too). Activate it with the "
-       (command-markup 'nyxt/visual-mode:visual-mode) " command.")
-   (:p "Visual mode provides the following commands: ")
+   (:p "Select text without a mouse. Turning on Nyxt's " (:code "visual-mode") "
+(with the " (command-markup 'nyxt/visual-mode:visual-mode) " command) imitates that
+of Vim's (and comes with the CUA and Emacs-like keybindings out of the box, too).")
```
Comment (aartaka) [inline source/tutorial.lisp:224]: "that of Vim"?

**Review by aartaka — COMMENTED** [review]
Thanks for this initiative -- manual have indeed gotten a bit rusty. Here are some points that hooked me.

**Comment by jmercouris** [comment]
I'm afraid that this pull request has become too stale. Much of the manual is now generated dynamically. I will close this pull request for now. Sorry for the disappointment. Thank you in any case!

**Pull request closed without merging**